app/services/settings/settings.ts

```diff
@@ -241,7 +199,44 @@ export class SettingsService extends StatefulService<ISettingsState>
       }
     });
 
-    return settingValue;
+    return inputModel;
+  }
+
+  findSettingValue(settings: ISettingsSubCategory[], category: string, setting: string) {
+    const formModel = this.findSetting(settings, category, setting);
+    if (!formModel) return;
+    return formModel.value !== void 0
+      ? formModel.value
+      : (formModel as IObsListInput<string>).options[0].value;
+  }
+
+  findValidListValue(settings: ISettingsSubCategory[], category: string, setting: string) {
+    const formModel = this.findSetting(settings, category, setting);
+    if (!formModel) return;
+    const options = (formModel as IObsListInput<string>).options;
+    const option = options.find(option => option.value === formModel.value);
+    return option ? option.value : options[0].value;
+  }
+
+  private patchSetting(
+    settingsFormData: ISettingsSubCategory[],
+    name: string,
+    patch: Partial<IObsInput<TObsValue>>,
+  ) {
+    // tslint:disable-next-line
+    settingsFormData = cloneDeep(settingsFormData);
```

Comment:
confused by this line

Reply:
Cloning prevents this object to be changed outside this function